cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -414,9 +415,15 @@ public boolean apply(ProgramRuntimeService.RuntimeInfo runtimeInfo) {
                                          "The following programs are still running: " + appAllRunningPrograms);
     }
     //All Apps are STOPPED, delete them
+    Set<String> appNames = new HashSet<>();
     for (ApplicationSpecification appSpec : allSpecs) {
-      ApplicationId id = new ApplicationId(namespaceId.getId(), appSpec.getName());
-      deleteApp(id, appSpec);
+      appNames.add(appSpec.getName());
+    }
+    for (String appName : appNames) {
+      Collection<ApplicationId> appIds = store.getAllAppVersionsAppIds(namespace, appName);
```

Comment:
That method will only return a collection of ApplicationSpecification's which does not contain version, but we need version here to construct appId